src/lightcurve.jl

```diff
@@ -0,0 +1,1243 @@
+including source information, timing parameters, and processing history.
+
+# Fields
+- `telescope::String`: Name of the telescope/mission
+- `instrument::String`: Name of the instrument
+- `object::String`: Name of the observed object
+- `mjdref::Float64`: Modified Julian Date reference time
+- `time_range::Tuple{Float64,Float64}`: Start and stop times of the light curve
+- `bin_size::Float64`: Time bin size in seconds
+- `headers::Vector{Dict{String,Any}}`: Original FITS headers
+- `extra::Dict{String,Any}`: Additional metadata and processing information
+
+# Examples
+```julia
+# Metadata is typically created automatically
+lc = create_lightcurve(eventlist, 1.0)
+println(lc.metadata.telescope)    # "NICER"
+println(lc.metadata.bin_size)     # 1.0
+println(lc.metadata.time_range)   # (1000.0, 2000.0)
+```
+"""
+struct LightCurveMetadata
+    "Name of the telescope/mission"
+    telescope::String
+    "Name of the instrument"
+    instrument::String
+    "Name of the observed object"
+    object::String
+    "Modified Julian Date reference time"
+    mjdref::Float64
+    "Start and stop times of the light curve"
+    time_range::Tuple{Float64,Float64}
+    "Time bin size in seconds"
+    bin_size::Float64
+    "Original FITS headers"
+    headers::Vector{Dict{String,Any}}
+    "Additional metadata and processing information"
+    extra::Dict{String,Any}
+end
+
+"""
+    LightCurve{T} <: AbstractLightCurve{T}
+
+A structure representing a binned time series with additional properties.
+
+This is the main light curve structure that holds binned photon count data
+along with statistical uncertainties, exposure times, and derived properties.
+
+# Fields
+- `timebins::Vector{T}`: Time bin centers
+- `bin_edges::Vector{T}`: Time bin edges (length = timebins + 1)
+- `counts::Vector{Int}`: Photon counts in each bin
+- `count_error::Vector{T}`: Statistical uncertainties on counts
+- `exposure::Vector{T}`: Exposure time for each bin
+- `properties::Vector{EventProperty}`: Additional computed properties
+- `metadata::LightCurveMetadata`: Comprehensive metadata
+- `err_method::Symbol`: Error calculation method used (:poisson or :gaussian)
+
+# Examples
+```julia
+# Create from event list
+ev = readevents("events.fits")
+lc = create_lightcurve(ev, 1.0)  # 1-second bins
+
+# Access data
+println("Time bins: ", lc.timebins[1:5])
+println("Counts: ", lc.counts[1:5])
+println("Errors: ", lc.count_error[1:5])
+
+# Basic operations
+println("Total counts: ", sum(lc.counts))
+println("Mean count rate: ", mean(lc.counts ./ lc.exposure))
+```
+
+# Interface
+- `length(lc)`: Number of time bins
+- `lc[i]`: Get (time, counts) tuple for bin i
+- Supports array-like indexing and iteration
+
+See also [`create_lightcurve`](@ref), [`rebin`](@ref).
+"""
+struct LightCurve{T} <: AbstractLightCurve{T}
+    "Time bin centers"
+    timebins::Vector{T}
+    "Time bin edges (length = timebins + 1)"
+    bin_edges::Vector{T}
```

Comment:
```suggestion
    dt::Union{T,Vector{T}}
```
When uniformly sampled, `dt` can be a single value. If non-uniformly sampled, this should be a vector.